Userland/chmod.cpp

```diff
@@ -3,18 +3,53 @@
 #include <sys/stat.h>
 #include <unistd.h>
 
+int str_to_octal_mode(char access_scope, char** access_str, unsigned& mode);
+int add_perm(char access_scope, char access_perm, char op_type, unsigned& mode);
```

Comment:
Let's call this function something more descriptive, like `apply_permission`.
Since it only returns success/failure, it should return a `bool` instead of an `int`.